tenseal/cpp/tensors/bfvvector.cpp

```diff
@@ -187,8 +227,17 @@ shared_ptr<BFVVector> BFVVector::dot_plain_inplace(
 }
 
 shared_ptr<BFVVector> BFVVector::sum_inplace(size_t /*axis=0*/) {
-    sum_vector(this->tenseal_context(), this->_ciphertext, this->size());
-    this->_size = 1;
+    vector<Ciphertext> interm_sum;
+    for (size_t idx = 0; idx < this->_ciphertexts.size(); ++idx) {
+        Ciphertext out = this->_ciphertexts[idx];
+        sum_vector(this->tenseal_context(), out, this->_sizes[idx]);
+        interm_sum.push_back(out);
+    }
```

Comment:
This can actually be computed in parallel. Please leave a TODO. Maybe after adding some helpers for parallel computations, then we can get back to this